src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -740,6 +740,29 @@ impl ExecutorContext {
                 )
             })?;
 
+        // Flush the batch with all the end commands.
+        // If we don't do this then some programs might not fully finish executing.
+        self.engine
+            .flush_batch(true, SourceRange::default())
```

Comment:
Pretty sure this needs to be inside `execute_and_build_graph()` so that any commands are captured in the artifact graph.

Reply:
oh shit fair

Reply:
oh hey thats. how i can test it!

Reply:
will do tomorrow my brain is fried today

Reply:
Nice! When I added the cube_with_error test, I thought it was odd that commands were lost by adding KCL code.